src/main/java/uk/gov/ons/ssdc/supporttool/schedule/RowChunkProcessor.java

```diff
@@ -41,42 +47,26 @@ public boolean processChunk(Job job) {
         job.getCollectionExercise().getSurvey().getSampleValidationRules();
 
     List<JobRow> jobRows =
-        jobRowRepository.findTop500ByJobAndAndJobRowStatus(job, JobRowStatus.STAGED);
+        jobRowRepository.findTop500ByJobAndAndJobRowStatus(job, JobRowStatus.VALIDATED_OK);
 
     for (JobRow jobRow : jobRows) {
-      JobRowStatus rowStatus = JobRowStatus.PROCESSED_OK;
-      List<String> rowValidationErrors = new LinkedList<>();
+      try {
+        ListenableFuture<String> future =
+            pubSubTemplate.publish(
+                sampleTopic, TRANSFORMER.transformRow(jobRow.getRowData(), job, columnValidators));
 
-      for (ColumnValidator columnValidator : columnValidators) {
-        Optional<String> columnValidationErrors = columnValidator.validateRow(jobRow.getRowData());
-        if (columnValidationErrors.isPresent()) {
-          rowStatus = JobRowStatus.PROCESSED_ERROR;
-          rowValidationErrors.add(columnValidationErrors.get());
-          hadErrors = true;
-        }
-      }
-
-      boolean retryBecausePubsubFailed = false;
-
-      if (rowValidationErrors.size() == 0) {
-        try {
-          pubSubTemplate.publish(
-              sampleTopic, TRANSFORMER.transformRow(jobRow.getRowData(), job, columnValidators));
-        } catch (Exception e) {
-          retryBecausePubsubFailed = true;
-
-          log.with(jobRow).error("Failed to send message to pubsub", e);
-        }
-      }
+        // Wait for up to 30 seconds to confirm that message was published
+        future.get(30, TimeUnit.SECONDS);
 
-      // If we had a problem with pubsub, don't bother changing the status of the row, so that we
-      // will keep retrying to send it, indefinitely
-      if (!retryBecausePubsubFailed) {
-        jobRow.setValidationErrorDescriptions(String.join(", ", rowValidationErrors));
-        jobRow.setJobRowStatus(rowStatus);
+        jobRow.setJobRowStatus(JobRowStatus.PROCESSED);
+        job.setProcessingRowNumber(job.getProcessingRowNumber() + 1);
+      } catch (Exception e) {
+        // The message sending will be retried...
+        log.with(jobRow).error("Failed to send message to pubsub", e);
```

Comment:
I think we should avoid logging the job row itself, might leak sample data/PII into the logs